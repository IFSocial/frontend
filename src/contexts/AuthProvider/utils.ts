import api from '../../services/api';
import { IUser } from './types';

// Salva o usuario no BD local
export function setUserLocalStorage(user: IUser | null) {
  localStorage.setItem('u', JSON.stringify(user));
}

// Checa se existe o usuario no BD local
export function getUserLocalStorage() {
  const json = localStorage.getItem('u');
  if (!json) {
    return null;
  }

  const user = JSON.parse(json);

  return user ?? null;
}

// Faz request do login passando email e senha como payload
export async function LoginRequest(email: string, password: string) {
  try {
    const request = await api.post('/users/signin', { email, password });

    return request.data;
  } catch {
    return null;
  }
}

// Faz request do cadastrp passando nome, email e senha como payload
export async function SignupRequest(
  name: string,
  registration: string,
  sexo: string,
  email: string,
  password: string,
) {
  try {
    const request = await api.post('/users/signup', {
      name,
      registration,
      sexo,
      email,
      password,
    });

    return request.data;
  } catch {
    return null;
  }
}
