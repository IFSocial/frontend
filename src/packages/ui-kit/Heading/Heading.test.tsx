import React from 'react';

import { Heading } from '.';
import { render } from '../../test/utils';

describe('Heading', () => {
  it('should render correctly', () => {
    const { getByRole } = render(<Heading>Heading</Heading>);

    const headingElement = getByRole('heading');

    expect(headingElement).toBeInTheDocument();
    expect(headingElement).toHaveTextContent(/^Heading$/);

    expect(headingElement).toMatchSnapshot();
  });

  it('should render a heading 1', () => {
    const { getByRole } = render(<Heading size={1}>Heading</Heading>);

    const headingElement = getByRole('heading');

    expect(headingElement).toBeInTheDocument();

    expect(headingElement).toMatchSnapshot();
  });

  it('should render a heading 2', () => {
    const { getByRole } = render(<Heading size={2}>Heading</Heading>);

    const headingElement = getByRole('heading');

    expect(headingElement).toBeInTheDocument();

    expect(headingElement).toMatchSnapshot();
  });

  it('should render a heading 3', () => {
    const { getByRole } = render(<Heading size={3}>Heading</Heading>);

    const headingElement = getByRole('heading');

    expect(headingElement).toBeInTheDocument();

    expect(headingElement).toMatchSnapshot();
  });

  it('should render a heading 4', () => {
    const { getByRole } = render(<Heading size={4}>Heading</Heading>);

    const headingElement = getByRole('heading');

    expect(headingElement).toBeInTheDocument();

    expect(headingElement).toMatchSnapshot();
  });

  it('should render a heading 5', () => {
    const { getByRole } = render(<Heading size={5}>Heading</Heading>);

    const headingElement = getByRole('heading');

    expect(headingElement).toBeInTheDocument();

    expect(headingElement).toMatchSnapshot();
  });

  it('should render a heading 6', () => {
    const { getByRole } = render(<Heading size={6}>Heading</Heading>);

    const headingElement = getByRole('heading');

    expect(headingElement).toBeInTheDocument();

    expect(headingElement).toMatchSnapshot();
  });
});
