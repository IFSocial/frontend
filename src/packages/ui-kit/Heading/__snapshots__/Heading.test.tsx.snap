// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render a heading 1 1`] = `
<h1
  class="sc-bcXHqe gKaYpH"
>
  Heading
</h1>
`;

exports[`Heading should render a heading 2 1`] = `
<h2
  class="sc-gswNZR gWRsUQ"
>
  Heading
</h2>
`;

exports[`Heading should render a heading 3 1`] = `
<h3
  class="sc-dkrFOg cZktjy"
>
  Heading
</h3>
`;

exports[`Heading should render a heading 4 1`] = `
<h4
  class="sc-hLBbgP cMErrE"
>
  Heading
</h4>
`;

exports[`Heading should render a heading 5 1`] = `
<h5
  class="sc-eDvSVe bOWnby"
>
  Heading
</h5>
`;

exports[`Heading should render a heading 6 1`] = `
<h6
  class="sc-jSUZER eUOdIN"
>
  Heading
</h6>
`;

exports[`Heading should render correctly 1`] = `
<h1
  class="sc-bcXHqe gKaYpH"
>
  Heading
</h1>
`;
